code/game/machinery/digital_clock.dm

```diff
@@ -0,0 +1,161 @@
+	icon_state = "digital_clock_base"
+	icon = 'icons/obj/digital_clock.dmi'
+	verb_say = "beeps"
+	verb_ask = "bloops"
+	verb_exclaim = "blares"
+	density = FALSE
+	layer = ABOVE_WINDOW_LAYER
+	custom_materials = list(/datum/material/iron= SHEET_MATERIAL_AMOUNT * 7, /datum/material/glass= SHEET_MATERIAL_AMOUNT * 4)
+
+/obj/item/wallframe/digital_clock
+	name = "digital clock frame"
+	desc = "Used to build digital clocks, just secure to the wall."
+	icon_state = "digital_clock_base"
+	icon = 'icons/obj/digital_clock.dmi'
+	custom_materials = list(/datum/material/iron= SHEET_MATERIAL_AMOUNT * 7, /datum/material/glass= SHEET_MATERIAL_AMOUNT * 4)
+	result_path = /obj/machinery/digital_clock
+	pixel_shift = 28
+
+/obj/machinery/digital_clock/wrench_act_secondary(mob/living/user, obj/item/tool)
+	. = ..()
+	balloon_alert(user, "[anchored ? "un" : ""]securing...")
+	tool.play_tool_sound(src)
+	if(tool.use_tool(src, user, 6 SECONDS))
+		playsound(loc, 'sound/items/deconstruct.ogg', 50, TRUE)
+		balloon_alert(user, "[anchored ? "un" : ""]secured")
+		deconstruct()
+		return TRUE
+
+/obj/machinery/digital_clock/welder_act(mob/living/user, obj/item/tool)
+	if(user.combat_mode)
+		return
+	if(atom_integrity >= max_integrity)
+		balloon_alert(user, "it doesn't need repairs!")
+		return TRUE
+	user.balloon_alert_to_viewers("repairing display...", "repairing...")
+	if(!tool.use_tool(src, user, 4 SECONDS, amount = 0, volume=50))
+		return TRUE
+	balloon_alert(user, "repaired")
+	atom_integrity = max_integrity
+	set_machine_stat(machine_stat & ~BROKEN)
+	update_appearance()
+	return TRUE
+
+/obj/machinery/digital_clock/multitool_act(mob/living/user, obj/item/tool)
+	if(user.combat_mode)
+		return
+	if(!(obj_flags & EMAGGED))
+		return
+	balloon_alert(user, "resetting")
+	tool.play_tool_sound(src)
+	if(tool.use_tool(src, user, 6 SECONDS))
+		playsound(loc, 'sound/items/deconstruct.ogg', 50, TRUE)
+		balloon_alert(user, "reset")
+		obj_flags &= ~EMAGGED
+		return TRUE
+
+/obj/machinery/digital_clock/emag_act(mob/user)
+	if(obj_flags & EMAGGED)
+		return
+	playsound(src, SFX_SPARKS, 100, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	do_sparks(3, FALSE, src)
+	obj_flags |= EMAGGED
+
+/obj/machinery/digital_clock/emp_act(severity)
+	. = ..()
+	emag_act()
+
+/obj/machinery/digital_clock/deconstruct(disassembled = TRUE)
+	if(flags_1 & NODECONSTRUCT_1)
+		return
+	if(!disassembled)
+		new /obj/item/stack/sheet/iron(drop_location(), 2)
+		new /obj/item/shard(drop_location())
+		new /obj/item/shard(drop_location())
+	else
+		new /obj/item/wallframe/digital_clock(drop_location())
+	qdel(src)
+
+//MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/digital_clock, 32)
+
+/obj/machinery/digital_clock/Initialize(mapload)
+	. = ..()
+	START_PROCESSING(SSdigital_clock, src)
+	update_appearance()
```

Comment:
`update_appearance()` should already be being called by `machinery/Initialize`, are you sure you need this?